docs/dev-guide/intellij-plugin.md

```diff
@@ -0,0 +1,187 @@
+| `textDocument/semanticTokens`              | ✅ Implemented        | Handled by lsp4ij     | Semantic syntax highlighting                                                                                                     |
+| `textDocument/references`                  | ✅ Implemented        | Handled by lsp4ij     | Find all references to a symbol                                                                                                  |
+| `textDocument/documentHighlight`           | ✅ partly implemented | Handled by lsp4ij     | Highlight related symbols; currently the highlight only works upon entirely selecting a symbol not just placing the carret there |
+| `textDocument/documentSymbol`              | ✅ Implemented        | Handled by lsp4ij     | Document outline/structure view                                                                                                  |
+| `textDocument/inlayHint`                   | ✅ Implemented        | Handled by lsp4ij     | Inlay additional information into code editor, i.e. the names of function parameters                                             |
+| `textDocument/codeAction`                  | ✅ Implemented        | Handled by lsp4ij     | Code fixes and refactoring actions                                                                                               |
+| `textDocument/formatting`                  | ❌ Not implemented    | -                     | Document formatting                                                                                                              |
+| `textDocument/rangeFormatting`             | ❌ Not implemented    | -                     | Range-based formatting                                                                                                           |
+| `textDocument/onTypeFormatting`            | ❌ Not implemented    | -                     | Format-on-type                                                                                                                   |
+| `textDocument/codeLens`                    | ❌ Not implemented    | -                     | Inline code annotations                                                                                                          |
+| `textDocument/foldingRange`                | ✅ Implemented        | Handled by lsp4ij     | Code folding regions                                                                                                             |
+| `textDocument/selectionRange`              | ✅ Implemented        | Handled by lsp4ij     | Smart text selection                                                                                                             |
+| `textDocument/prepareCallHierarchy`        | ❌ Not implemented    | -                     | Call hierarchy preparation                                                                                                       |
```

Comment:
Lsp4j can support it, why you  say not implemented ?

Reply:
prepareCallHierarchy also did not work in my experiments, however im not sure, where its failing